internal/pkg/cfa/testdata/src/example.com/tests/tests.go

```diff
@@ -0,0 +1,88 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cfa
```

Comment:
All kinds of cycles are handled now, though I suspect there are cases where the analysis may be incomplete or too liberal (biased towards false positives). I think ultimately handling cycles properly may require us to discover cycles in advance and perform several rounds of analysis on each cycle, e.g. in a first pass you do not traverse through calls to a function in the cycle, then you re-do the analysis using the knowledge acquired in the first pass, and you repeat until you reach a fixed point. This may not be the case though, a future PR will be needed to clear this up.